corona-warn-app/cwa-app-android
# Reduce code duplication (EXPOSUREAPP-2844)

## Description
Previously we had two `MainFragment` where the only different were the menu entries to the 2 test fragments we have. This also duplicated the `nav_graph.xml` in turn it caused by subtle issues and cost us time when switching between variants caused build errors until cache was cleared and where features were added to only one of the fragments.

This PR removes the extra `MainFragment`, introduces a `TestMenuFragment` as root for all test related screens, and hooks that into the main `nav_graph` via nested nav-graph inclusion. In `device` flavor builds, the nested nav-graph is a stub (empty) and the menu entry is hidden.

Future PRs can refactor the existing test fragments and move the mixed test code into their own Fragments to offer improved modularity and testing.

## How to test
* Review code etc.
* Deploy both `device` and `deviceForTesters` variant, the former should not have a test menu, while the later should have one.
* Navigate the test menu
